plugin/src/main/java/org/gradle/testretry/internal/RetryTestResultProcessor.java

```diff
@@ -84,7 +93,9 @@ public void output(Object testId, TestOutputEvent testOutputEvent) {
     public void failure(Object testId, Throwable throwable) {
         TestDescriptorInternal descriptor = activeDescriptorsById.get(testId);
         if (descriptor != null && descriptor.getClassName() != null) {
-            failedTests.add(new TestName(descriptor.getClassName(), descriptor.getName()));
+            final TestName test = new TestName(descriptor.getClassName(), descriptor.getName());
+            nonExecutedFailedTests.remove(test);
```

Comment:
❓ Why are we removing it here and in `completed`?

Reply:
previously it was executed beforehand in the `started` method which is called on every test execution. 
https://github.com/gradle/test-retry-gradle-plugin/pull/59/files#diff-c3de119e7a9886810b316ca4173f467eL55
But you are right its needless in the FAILED case,